File: java/src/test/java/problems201_300/ValidAnagramTest.java
package problems201_300;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidAnagramTest extends SampledTest {
    @Test
    public void test() {
        for (ValidAnagram solution : ValidAnagram.values()) {
            for (ValidAnagramSample sample : getSamples(ValidAnagramSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isAnagram(sample.input.s, sample.input.t))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/ValidAnagramTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ValidAnagramTest : SampledTest() {
    @Test
    fun test() =
        ValidAnagram.entries.forEach { solution ->
            getSamples<Array<ValidAnagramSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.isAnagram(input.s, input.t))
                    .isEqualTo(output)
            }
        }
}
